device/bgp/bgpsrtesegment.yaml

```diff
@@ -3,47 +3,490 @@ components:
+            Type  C: IPv4 Node Address with optional SID.
+            Type  D: IPv6 Node Address with optional SID for SR MPLS.
+            Type  E: IPv4 Address and index with optional SID.
+            Type  F: IPv4 Local and Remote addresses with optional SID.
+            Type  G: IPv6 Address and index for local and remote pair with optional
+                  SID for SR MPLS.
+            Type  H: IPv6 Local and Remote addresses with optional SID for SR MPLS.
+            Type  I: IPv6 Node Address with optional SID for SRv6.
+            Type  J: IPv6 Address and index for local and remote pair with optional
+                  SID for SRv6.
+            Type  K: IPv6 Local and Remote addresses for SRv6.
           type: string
           enum:
-          - mpls_sid
-          - ipv6_sid
-          default: mpls_sid
-        mpls_label:
+          - type_a
+          - type_b
+          - type_c
+          - type_d
+          - type_e
+          - type_f
+          - type_g
+          - type_h
+          - type_i
+          - type_j
+          - type_k
+        type_a:
+          $ref: '#/components/schemas/BgpSrte.SegmentATypeSubTlv'
+        type_b:
+          $ref: '#/components/schemas/BgpSrte.SegmentBTypeSubTlv'
+        type_c:
+          $ref: '#/components/schemas/BgpSrte.SegmentCTypeSubTlv'
+        type_d:
+          $ref: '#/components/schemas/BgpSrte.SegmentDTypeSubTlv'
+        type_e:
+          $ref: '#/components/schemas/BgpSrte.SegmentETypeSubTlv'
+        type_f:
+          $ref: '#/components/schemas/BgpSrte.SegmentFTypeSubTlv'
+        type_g:
+          $ref: '#/components/schemas/BgpSrte.SegmentGTypeSubTlv'
+        type_h:
+          $ref: '#/components/schemas/BgpSrte.SegmentHTypeSubTlv'
+        type_i:
+          $ref: '#/components/schemas/BgpSrte.SegmentITypeSubTlv'
+        type_j:
+          $ref: '#/components/schemas/BgpSrte.SegmentJTypeSubTlv'
+        type_k:
+          $ref: '#/components/schemas/BgpSrte.SegmentKTypeSubTlv'
+  
+    BgpSrte.SrMplsSid:
+      description: >-
+        Configuration for SR-MPLS with Label, TC, Bottom-of-Stack and TTL.
+      type: object
+      properties:
+        label:
           description: >-
-            MPLS label
+            Label value in [0, 2^20 -1].
           type: integer
-          default: 0  
-        mpls_tc:
+          minimum: 0
+          maximum: 1048575
+        tc:
           description: >-
-            TBD
+            Traffic class in bits.
           type: integer
+          minimum: 0
+          maximum: 7
+        s_bit:
+          description: >-
+            Bottom-of-Stack bit.
+          type: boolean
+        ttl:
+          description: >-
+            Time To Live. 
+          type: integer
+          minimum: 0
+          maximum: 255
+
+    BgpSrte.SRv6SIDEndpointBehaviorAndStructure:
+      description: >-
+        SRv6 Endpoint Behavior and SID Structure and its optional. 
```

Comment:
Configuration for SRv6 Endpoint Behavior and SID Structure. Its optional. Summation of lengths for Locator Block, Locator Node, Function, and Argument  MUST be less than or equal to 128.